File: LocaleConnectBackend/trip/src/main/java/com/localeconnect/app/trip/model/Trip.java
package com.localeconnect.app.trip.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "trip")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Trip {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name = "localguide_id", nullable = false, unique = true)
    private Long localguideId;
    @Column(nullable = false, unique = true)
    private String name;
    private String description;
    private LocalDate departureTime;
    private String destination;
    private Double travelTime;
    private Integer capacity;

    @ElementCollection
    private List<Long> travelers = new ArrayList<>();

    @ElementCollection
    private List<String> languages = new ArrayList<>();

    @ElementCollection
    private List<String> dailyActivities = new ArrayList<>();

    @ElementCollection
    private List<String> imageUrls = new ArrayList<>();

    @Column
    private double ratingsTotal;

    @Column
    private int ratingsCount;

    @Column
    private double averageRating;
    public void addRating(double rating) {
        this.ratingsTotal += rating;
        this.ratingsCount++;
    }
    public void calcAverageRating() {
        this.averageRating = (this.ratingsCount > 0) ? this.ratingsTotal / this.ratingsCount : 0;
    }


}

File: LocaleConnectFrontend/src/app/model/trip.ts

export interface TripPreview {
  id: string;
  name: string;
  description: string;
  link: string;
}
export interface Trip {
  localguideId?: number;
  name: string;
  description?: string;
  departureTime: string;
  destination: string;
  durationInDays: number;
  capacity: number;
  travelers: number[];
  languages: string[];
  placesToVisit: string[];
  dailyActivities?: string[];
  imageUrls?: string[];
  rating?: number;
  ratingSubmitted?: boolean;
  totalRatings?: number;
  averageRating?: number;
  expand?: boolean;
}

export interface TripReview {
  tripReviewId?: number;
  tripId: number;
  userId: number;
  text?: string;
  timestamp?: string;
  rating?: number;
}

export interface TripShare {
  id?: number;
  name: string;
  description?: string;
}
